src/Sparrow/Utils/MemoryUtils.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Linq;
+using Sparrow.LowMemory;
+
+namespace Sparrow.Utils;
+
+public static class MemoryUtils
+{
+    public static string GetExtendedMemoryInfo(MemoryInfoResult memoryInfo)
+    {
+        var message =
```

Comment:
Also, let's use a string builder to reduce allocations

Reply:
done